### PR TITLE
fix(java): grow buffer to interger.max

### PR DESCRIPTION
**Pull request by chaokunyang**
## What does this PR do?

grow buffer to interger.max

## Related issues

#1576 

## Does this PR introduce any user-facing change?

- [ ] Does this PR introduce any public API change?
- [ ] Does this PR introduce any binary protocol compatibility change?

### CODE REVIEW

File: java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java [hunk 1]
```diff
@@ -1204,18 +1204,27 @@ public void writePrimitiveArray(Object arr, int offset, int numBytes) {
...
-      byte[] data = new byte[length * 2];
-      copyToUnsafe(0, data, Platform.BYTE_ARRAY_OFFSET, size());
-      initHeapBuffer(data, 0, data.length);
+      growBuffer(length);
     }
   }
 
+  private void growBuffer(int length) {
+    int newSize =
+        length < 104857600 ? length << 2 : (int) Math.min(length * 1.5d, Integer.MAX_VALUE);
```
Comment (LiangliangSui) [inline java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java:1222]: How is the value `104857600` calculated?

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java:1222]: It' just 100 * 1024 * 1024

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java:1222]: Move `org.apache.fury.io.FuryStreamReader#BUFFER_GROW_STEP_THRESHOLD` to `MemoryBuffer` and reuse it?

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java]: good suggestion, updated

**Review by LiangliangSui — APPROVED** [review]
LGTM, thanks!

**Pull request merged**